cogs/fun.py

```diff
@@ -1,16 +1,231 @@
 import random
 import textwrap
+import urllib
 
+import aiohttp
 import discord
 from discord import Color, Embed
 from discord.ext.commands import BadArgument, Bot, BucketType, Cog, Context, command, cooldown
 
+from cogs.utils.errors import ServiceError
+
+from .endpoints.endpoints import nekos
+from .utils.plainreq import get_req
+
 
 class Fun(Cog):
     """This is a cog for simple fun commands."""
 
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
+        self.user_agent = {"User-Agent": "HotWired"}
+        self.dadjoke = {
+            "User-Agent": "HotWired",
+            "Accept": "text/plain",
+        }
+
+    @command()
+    async def cat(self, ctx: Context) -> None:
+        """Random cat images. Awww, so cute! Powered by random.cat"""
+        async with self.bot.session.get("https://aws.random.cat/meow", headers=self.user_agent) as r:
+            if r.status == 200:
+                js = await r.json()
+                em = discord.Embed(name="random.cat", colour=0x690E8)
+                em.set_image(url=js["file"])
+                await ctx.send(embed=em)
+            else:
+                raise ServiceError(f"could not fetch cute cat :( (http {r.status})")
+
+    @command()
+    async def httpcat(self, ctx: Context, http_id: int) -> None:
+        """http.cat images - ^httpcat <http code>"""
+        codes = [
+            100,
+            101,
+            200,
+            201,
+            202,
+            204,
+            206,
+            207,
+            300,
+            301,
+            302,
+            303,
+            304,
+            305,
+            307,
+            400,
+            401,
+            402,
+            403,
+            404,
+            405,
+            406,
+            408,
+            409,
+            410,
+            411,
+            412,
+            413,
+            414,
+            416,
+            417,
+            418,
+            420,
+            421,
+            422,
+            423,
+            424,
+            425,
+            426,
+            429,
+            444,
+            450,
+            451,
+            500,
+            502,
+            503,
+            504,
+            506,
+            507,
+            508,
+            509,
+            511,
+            599,
+        ]
```

Comment:
This is unacceptable, is there really no better way to check it?
Even if there isn't, you can at least put more error codes on single line, this is taking up way too many lines. 
You can do that like this:
```py
codes = [
    100, 101, 200, 201, 204, 206, 207,
    300, 301, 302, 303. 304. 305, 307,
    ...
]
```

Reply:
Keep this as pending, I'll probably remove black all togeter, also make sure to post your replies here, not as new comments, while marking this as resolved

Reply:
Sure, 👍